JASP-Engine/JASP/R/binomialtest.R

```diff
@@ -15,224 +15,315 @@
+}
 
-  if (perform == "init" || is.null(variables)) {
+# Results functions ----
+.binomComputeResults <- function(jaspResults, dataset, options) {
+
+  # Take results from state if possible
+  if (!is.null(jaspResults[["stateBinomResults"]])) return(jaspResults[["stateBinomResults"]]$object)
+
+  # This will be the object that we fill with results
+  results <- list()
+
+  # First, we perform precalculation of variables we use throughout the analysis
+  results$spec <- .binomCalcSpecs(dataset, options)
+
+  for (variable in options$variables) {
+
+    results[[variable]] <- list()
+
+    # Prepare for running the binomial test
+    column <- dataset[[.v(variable)]]
+    data   <- column[!is.na(column)]
+    levels <- levels(as.factor(data))
+
+    for (level in levels) {
+      nObs   <- length(data)
+      counts <- sum(data == level)
+      prop   <- counts / nObs
+
+      # Main binomial test result for tables
+      tableResults <- stats::binom.test(
+        x           = counts,
+        n           = nObs,
+        p           = options$testValue,
+        alternative = results$spec$hypothesisRec,
+        conf.level  = options$confidenceIntervalInterval
+      )
+
+      # sometimes p.value becomes true or false, convert this to 1 or 0
+      p <- as.numeric(tableResults$p.value)
+      lowerCI <- tableResults$conf.int[1]
+      upperCI <- tableResults$conf.int[2]
+
+      # Binomial test for plots
+      plotResults <- stats::binom.test(
+        x           = counts,
+        n           = nObs,
+        p           = options$testValue,
+        alternative = "two.sided",
+        conf.level  = options$descriptivesPlotsConfidenceInterval
+      )
+
+      # Summary statistics for plots
+      plotDat <- data.frame(
+        label      = level,
+        proportion = prop,
+        lowerCI    = plotResults$conf.int[1],
+        upperCI    = plotResults$conf.int[2]
+      )
+
+      # Add results for each level of each variable to results object
+      results[[variable]][[level]] <- list(
+        variable      = variable,
+        level         = level,
+        counts        = counts,
+        total         = nObs,
+        proportion    = prop,
+        p             = p,
+        VovkSellkeMPR = .VovkSellkeMPR(p),
+        lowerCI       = lowerCI,
+        upperCI       = upperCI,
+        plotDat       = plotDat
+      )
+    }
+  }
 
-    binomResults <- rbind(binomResults, rep(".", ncol(binomResults)))
+  # Save results to state
+  jaspResults[["stateBinomResults"]] <- createJaspState(results)
+  jaspResults[["stateBinomResults"]]$dependOnOptions(
+    c("variables", "testValue", "hypothesis", "confidenceIntervalInterval", "descriptivesPlotsConfidenceInterval")
+  )
 
+  # Return results object
+  return(results)
+}
+
+.binomCalcSpecs <- function(dataset, options) {
+  specs <- list()
+
+  # Recode the hypothesis for binom.test()
+  if (options$hypothesis == "notEqualToTestValue") {
```

Comment:
probably something to deal with later, but I see this particular if statement in almost every analysis..

Reply:
yes, ideally this would just be passed to the options directly as-is. I had a short conversation about this with @JorisGoosen. I think the argument @Janto91 gave is that [the `json` does cannot handle names with periods in them as in `two.sided`](https://github.com/jasp-stats/jasp-desktop/blob/development/Docs/development/r-analyses-guide.md#step-12-updating-the-options). This seems strange and fixable though!

Reply:
It seems that this isn't actually a problem in the json, but in the UI files (for the analysis widgets) where you give the name of an option as name of a widget.

And you cannot put a dot in the name because it is converted to c++. 
We could make this better in QML though, maybe. Although @boutinb would be able to answer this more authoritatively.